src/strategies/multichain/index.ts

```diff
@@ -0,0 +1,82 @@
+import {subgraphRequest} from '../../utils';
+import networks from '../../networks.json';
+import {JsonRpcProvider} from "@ethersproject/providers";
+import strategies from '..';
+
+export const author = 'kesarito';
+export const version = '1.0.0';
+
+async function getChainBlockNumber(timestamp: number, graphURL: string): Promise<number> {
+  const query = {
+    blocks: {
+      __args: {
+        first: 1,
+        orderBy: 'number',
+        orderDirection: 'desc',
+        where: {
+          timestamp_lte: timestamp
+        }
+      },
+      number: true,
+      timestamp: true
+    }
+  };
+  const data = await subgraphRequest(graphURL, query);
+  return Number(data.blocks[0].number);
+}
+
+async function getChainBlocks(snapshot, provider, options, network): Promise<any> {
+
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const block = await provider.getBlock(blockTag);
+  const chainBlocks = {};
+  for (let strategy of options.strategies) {
+    if (chainBlocks[strategy.network]) {
+      continue;
+    }
+    if (strategy.network === network) {
+      chainBlocks[strategy.network] = blockTag;
+    } else {
+      chainBlocks[strategy.network] = await getChainBlockNumber(block.timestamp, options.graphs[strategy.network]);
```

Comment:
If the snapshot is "latest" you actually don't need to get the corresponding block number on others chain you can just use "latest" as snapshot and the strategies will be resolved at the current networks state.